csrc/scheduler/normalization_utils.cpp

```diff
@@ -722,5 +722,26 @@ getOptionalInnerOuterPersistentBufferBatches(
   }
 }
 
+TensorView* getReferenceReductionTv(
+    const std::vector<TensorView*>& reduction_tvs) {
+  TensorView* first_inner_tv = nullptr;
+  TensorView* first_outer_tv = nullptr;
+  for (auto tv : reduction_tvs) {
+    bool is_inner = scheduler_utils::isFastestDimReduction(tv);
+
+    if (is_inner && !first_inner_tv) {
+      first_inner_tv = tv;
+    } else if (!is_inner && !first_outer_tv) {
+      first_outer_tv = tv;
+    }
+
+    if (first_inner_tv && first_outer_tv) {
+      return first_inner_tv;
+    }
+  }
+
+  return reduction_tvs[0];
```

Comment:
Can this tensor be an outer reduction tensor?

Reply:
Oh, I thought this function is only used for the combined case, but it's actually used in the other normalization cases. 

This seems to be a clear motivating example why these normalization schedulers should be decomposed.

Reply:
Yes, this is a common util function used by all kinds of normalization schedulers.